### PR TITLE
[20043] Moved all typelookupservice files to src

### PR DESCRIPTION
**Pull request by adriancampo**
This PR moves all TypeLookup service files from the "include" directory to "src"

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Changes are ABI compatible. 
- N/A Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [N/A] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -117,11 +117,11 @@ set(${PROJECT_NAME}_source_files
...
-    fastdds/builtin/typelookup/common/TypeLookupTypes.cpp
-    fastdds/builtin/common/RPCHeadersImpl.cpp
-    fastdds/builtin/typelookup/TypeLookupManager.cpp
-    fastdds/builtin/typelookup/TypeLookupRequestListener.cpp
-    fastdds/builtin/typelookup/TypeLookupReplyListener.cpp
+    fastdds/builtin/typelookupservice/common/TypeLookupTypes.cpp
+    fastdds/builtin/typelookupservice/common/RPCHeadersImpl.cpp
+    fastdds/builtin/typelookupservice/TypeLookupManager.cpp
+    fastdds/builtin/typelookupservice/TypeLookupRequestListener.cpp
+    fastdds/builtin/typelookupservice/TypeLookupReplyListener.cpp
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/CMakeLists.txt:124]: If we are going to rename the folder, then I suggest the following name: `type_lookup_service`

File: src/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -117,11 +117,11 @@ set(${PROJECT_NAME}_source_files
     fastdds/domain/DomainParticipant.cpp
     fastdds/domain/qos/DomainParticipantQos.cpp
     fastdds/domain/qos/DomainParticipantFactoryQos.cpp
-    fastdds/builtin/typelookup/common/TypeLookupTypes.cpp
-    fastdds/builtin/common/RPCHeadersImpl.cpp
-    fastdds/builtin/typelookup/TypeLookupManager.cpp
-    fastdds/builtin/typelookup/TypeLookupRequestListener.cpp
-    fastdds/builtin/typelookup/TypeLookupReplyListener.cpp
+    fastdds/builtin/typelookupservice/common/TypeLookupTypes.cpp
+    fastdds/builtin/typelookupservice/common/RPCHeadersImpl.cpp
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/CMakeLists.txt:121]: The folder `common` is going to disappear, leaving the types in the same folder. This can be done in the follow-up PR that generates the types from the IDL.

File: src/cpp/fastdds/builtin/typelookupservice/TypeLookupManager.hpp [hunk 3]
```diff
@@ -24,9 +24,9 @@
...
 
-#include <fastdds/dds/builtin/typelookup/TypeLookupRequestListener.hpp>
-#include <fastdds/dds/builtin/typelookup/TypeLookupReplyListener.hpp>
-#include <fastdds/dds/builtin/typelookup/common/TypeLookupTypes.hpp>
+#include <fastdds/builtin/typelookupservice/TypeLookupRequestListener.hpp>
+#include <fastdds/builtin/typelookupservice/TypeLookupReplyListener.hpp>
+#include <fastdds/builtin/typelookupservice/common/TypeLookupTypes.hpp>
 
 #include <fastdds/rtps/builtin/data/WriterProxyData.h>
 #include <fastdds/rtps/builtin/data/ReaderProxyData.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/typelookupservice/TypeLookupManager.hpp:32]: Please, follow the recommended including order. As these headers are private now, they should be included the latest, sorted alphabetically with the other internal header files. Please, apply this suggestion to the other files.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
Just a suggestion about the folder name

**Comment by JLBuenoLopez-eProsima** [comment]
@adriancampo, please,

* [ ] Fill the contributor checklist. If any check does not apply to this PR, just annotate it with N/A.
* [ ] Fix the conflicts
* [ ] Look at the CI warnings

I think this PR should be directed to the intermediate branch `feature/xtypes1.3` instead of creating a new intermediate branch.

File: src/cpp/CMakeLists.txt (hunk 2 above)
Comment (adriancampo) [reply src/cpp/CMakeLists.txt:121]: I think it makes sense to move all the related files together now, and then remove them in the idl's PR

File: src/cpp/fastdds/builtin/typelookupservice/TypeLookupManager.hpp (hunk 3 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/typelookupservice/TypeLookupManager.hpp]: Fixed in [2bfaa40](https://github.com/eProsima/Fast-DDS/pull/4053/commits/2bfaa40548d3668e183c3d0a84e25647ea09675e)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 4]
```diff
@@ -40,6 +38,7 @@
 
 #include <fastdds/dds/log/Log.hpp>
 
+#include <fastdds/builtin/type_lookup_service/TypeLookupManager.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:41]: This is the only exception. If the header file is the one defining the class that is implemented in the specific source file, then, it should be the first to be included. Please revert the change in this file.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 5]
```diff
@@ -24,13 +24,13 @@
...
-#include <fastdds/builtin/typelookupservice/TypeLookupRequestListener.hpp>
-#include <fastdds/builtin/typelookupservice/TypeLookupReplyListener.hpp>
-#include <fastdds/builtin/typelookupservice/common/TypeLookupTypes.hpp>
-
 #include <fastdds/rtps/builtin/data/WriterProxyData.h>
 #include <fastdds/rtps/builtin/data/ReaderProxyData.h>
 
+#include <fastdds/builtin/type_lookup_service/common/TypeLookupTypes.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupReplyListener.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:32]: It should be sorted alphabetically. Please, apply in the other files.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp [hunk 6]
```diff
@@ -17,8 +17,8 @@
  *
  */
 
-#include <fastdds/builtin/typelookupservice/TypeLookupReplyListener.hpp>
-#include <fastdds/builtin/typelookupservice/TypeLookupManager.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupReplyListener.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupManager.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:21]: These are private and have not been moved to the private headers block.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp [hunk 7]
```diff
@@ -29,6 +26,10 @@
 
 #include <utility>
 
+#include <fastdds/builtin/type_lookup_service/common/TypeLookupTypes.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp>
+#include <fastdds/builtin/type_lookup_service/TypeLookupManager.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp:31]: Please, apply recommended include order for every header included in this file:

1. Header declaring the implemented class. Blank line.
2. C headers (if possible avoid their use including the following). Blank line.
3. C++ headers. Blank line.
4. External header files (external dependencies). Blank line.
5. Public internal headers. Blank line.
6. Private internal headers.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp (hunk 4 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Fixed in [7cbeaf0](https://github.com/eProsima/Fast-DDS/pull/4053/commits/7cbeaf000b9780d2985f5b663a0244701a6b5cbe)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp (hunk 5 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Fixed in [7cbeaf0](https://github.com/eProsima/Fast-DDS/pull/4053/commits/7cbeaf000b9780d2985f5b663a0244701a6b5cbe)

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 8]
```diff
@@ -17,32 +17,33 @@
...
+#include <fastdds/rtps/history/WriterHistory.h>
+#include <fastdds/rtps/reader/RTPSReader.h>
+#include <fastdds/rtps/reader/StatefulReader.h>
+#include <fastdds/rtps/writer/RTPSWriter.h>
+#include <fastdds/rtps/writer/StatefulWriter.h>
 // TODO Uncomment if security is implemented.
 //#include <fastdds/rtps/common/Guid.h>
 //#include <fastdds/rtps/security/accesscontrol/ParticipantSecurityAttributes.h>
 
 #include <fastdds/dds/log/Log.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:43]: Nitpick: This header is public and should be included in the previous block

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp [hunk 9]
```diff
@@ -18,17 +18,18 @@
...
 
 #include <fastdds/builtin/type_lookup_service/TypeLookupReplyListener.hpp>
-#include <fastdds/builtin/type_lookup_service/TypeLookupManager.hpp>
-#include <fastdds/rtps/participant/RTPSParticipantListener.h>
-#include <fastdds/rtps/builtin/BuiltinProtocols.h>
 
+#include <fastrtps/rtps/history/ReaderHistory.h>
+#include <fastrtps/rtps/reader/RTPSReader.h>
 #include <fastrtps/types/TypeObjectFactory.h>
+#include <rtps/participant/RTPSParticipantImpl.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:25]: This header is private so it should be included in the block below.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp [hunk 10]
```diff
@@ -18,17 +18,18 @@
...
 
-#include <fastrtps/rtps/reader/RTPSReader.h>
-#include <fastrtps/rtps/history/ReaderHistory.h>
 #include <fastdds/dds/log/Log.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:27]: This header is public and should be included in the block above (sorted alphabetically).

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp [hunk 11]
```diff
@@ -18,17 +18,18 @@
...
 
-#include <rtps/participant/RTPSParticipantImpl.h>
+#include <fastdds/rtps/builtin/BuiltinProtocols.h>
+#include <fastdds/rtps/participant/RTPSParticipantListener.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:30]: These headers are public and should be in the block above

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp [hunk 12]
```diff
@@ -17,17 +17,17 @@
...
+#include <utility>
 
-#include <fastrtps/rtps/reader/StatefulReader.h>
 #include <fastrtps/rtps/history/ReaderHistory.h>
-#include <fastdds/dds/log/Log.hpp>
+#include <fastrtps/rtps/reader/StatefulReader.h>
+#include <fastrtps/types/TypeObjectFactory.h>
 
-#include <utility>
+#include <fastdds/dds/log/Log.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp:28]: This public header should be in the block above.

Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp]: [f288915](https://github.com/eProsima/Fast-DDS/pull/4053/commits/f288915aa343eabd6476d3875ea51d994ebe899d)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp (hunk 11 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:24]: [f288915](https://github.com/eProsima/Fast-DDS/pull/4053/commits/f288915aa343eabd6476d3875ea51d994ebe899d)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp (hunk 10 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp:22]: [f288915](https://github.com/eProsima/Fast-DDS/pull/4053/commits/f288915aa343eabd6476d3875ea51d994ebe899d)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp (hunk 9 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp]: [f288915](https://github.com/eProsima/Fast-DDS/pull/4053/commits/f288915aa343eabd6476d3875ea51d994ebe899d)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp (hunk 8 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: [f288915](https://github.com/eProsima/Fast-DDS/pull/4053/commits/f288915aa343eabd6476d3875ea51d994ebe899d)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test mac

**Comment by JLBuenoLopez-eProsima** [comment]
Mac CI failure is unrelated.
